akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/EventSourcedBehaviorStashSpec.scala

```diff
@@ -489,40 +489,38 @@ class EventSourcedBehaviorStashSpec
     "discard when stash has reached limit with default dropped setting" in {
       val probe = TestProbe[AnyRef]()
       system.toUntyped.eventStream.subscribe(probe.ref.toUntyped, classOf[Dropped])
-      val behavior = EventSourcedBehavior[String, String, Boolean](
-        persistenceId = PersistenceId("stash-is-full-drop"),
-        emptyState = false,
-        commandHandler = { (state, command) =>
-          state match {
-            case false =>
-              command match {
-                case "ping" =>
-                  probe.ref ! "pong"
-                  Effect.none
-                case "start-stashing" =>
-                  Effect.persist("start-stashing")
-                case msg =>
-                  probe.ref ! msg
-                  Effect.none
-              }
-
-            case true =>
-              command match {
-                case "unstash" =>
-                  Effect
-                    .persist("unstash")
-                    .thenUnstashAll()
-                    // FIXME #26489: this is run before unstash, so not sequentially as the docs say
-                    .thenRun(_ => probe.ref ! "done-unstashing")
-                case _ =>
-                  Effect.stash()
-              }
-          }
-        }, {
-          case (_, "start-stashing") => true
-          case (_, "unstash")        => false
-          case (_, _)                => throw new IllegalArgumentException()
-        })
+      val behavior = Behaviors.setup[String] { context =>
+        EventSourcedBehavior[String, String, Boolean](
+          persistenceId = PersistenceId("stash-is-full-drop"),
+          emptyState = false,
+          commandHandler = { (state, command) =>
+            state match {
+              case false =>
+                command match {
+                  case "ping" =>
+                    probe.ref ! "pong"
+                    Effect.none
+                  case "start-stashing" =>
+                    Effect.persist("start-stashing")
+                  case msg =>
+                    probe.ref ! msg
+                    Effect.none
+                }
+
+              case true =>
+                command match {
+                  case "unstash" =>
+                    Effect.persist("unstash").thenRun((_: Boolean) => context.self ! "done-unstashing").thenUnstashAll()
```

Comment:
here is the change, sending it to self so that it forwarded to the probe after all stashed